### PR TITLE
common functionality

### PR DESCRIPTION
**Pull request by Tumumanye-Nelson**
## Description of what I changed
1. Update Person to extend BaseCustomizableData
2. Update PersonAttribute to extend BaseAttribute
3. Update PersonAttributeType to extend BaseAttributeType

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-4231

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/Person.java [hunk 1]
```diff
@@ -45,7 +46,7 @@
  * @see org.openmrs.Patient
  */
 @Audited
-public class Person extends BaseChangeableOpenmrsData {
+public class Person extends BaseCustomizableData<PersonAttribute> implements Customizable<PersonAttribute>{
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/Person.java:49]: What are you trying to do here. Why are you extending  `BaseCustomizableData<PersonAttribute> implements Customizable<PersonAttribute>` ?

File: api/src/main/java/org/openmrs/Person.java [hunk 2]
```diff
@@ -970,6 +971,7 @@ public User getPersonChangedBy() {
 		return personChangedBy;
 	}
 	
+	@SuppressWarnings("deprecation")
```
Comment (sherrif10) [inline api/src/main/java/org/openmrs/Person.java:974]: i dont think this change makes sense.

**Comment by sherrif10** [comment]
This might be of help  .https://openmrs.atlassian.net/wiki/spaces/docs/pages/25477199/Pull+Request+Tips

File: api/src/main/java/org/openmrs/Person.java (hunk 1 above)
Comment (Tumumanye-Nelson) [reply api/src/main/java/org/openmrs/Person.java:49]: Before i had an error but i have solved it by allowing PersonAttributeType to extend BaseAttributeType.

what i want to achieve is to enable encapsulation, code reusability, inheritance, polymorphism, and abstraction, which together foster a modular, maintainable, and scalable codebase. These principles not only streamline the development process but also ensure that applications are robust, flexible, and easier to manage and extend.

**Pull request closed without merging**